src/sql/workbench/parts/dataExplorer/electron-browser/nodeActions.contribution.ts

```diff
@@ -19,34 +18,24 @@ MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	when: NodeContextKey.IsConnected
 });
 
-// The weird regex is because we want this to generically apply to Database and Server nodes but right now
-// there isn't a consistent standard for the values there. We can't just search for database or server being
-// in the string at all because there's lots of node types which have those values (such as ServerLevelLogin)
-// that we don't want these menu items showing up on.
 MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	group: 'connection',
 	order: 2,
 	command: {
 		id: NEW_QUERY_COMMAND_ID,
 		title: localize('newQuery', 'New Query')
 	},
-	when: ContextKeyExpr.and(
-		NodeContextKey.IsConnectable,
-		new ContextKeyRegexExpr('viewItem', /.+itemType\.database.*|^database$/i))
+	when: NodeContextKey.IsConnectable
 });
 
-// Note that we don't show this for Databases under Server nodes (viewItem == Database) because
-// of an issue there where the connection always being master instead of the actual DB
```

Comment:
With your changes that means this comment is no longer true, correct? So if you right click on a DB node and select manage it'll open the manage tab but with the context set to master instead of the DB. That's a separate issue but you should keep this behavior (hide manage item in those cases) if that's still the case

Reply:
Yes